### PR TITLE
Various fixes in docstrings

### PR DESCRIPTION
**Pull request by sofianehaddad**
* NumericalScalar --> Scalar
 * LinearModelResult is member of the otlm, not openturns
 * Fix LinearModelAnalysis::getNormalityTestResultChiSquared docstring :
 The class have been updated and the above method relies on the new
 FittingTest_ChiSquared without updating the docstrings

### CODE REVIEW

File: python/src/LinearModelAnalysis_doc.i.in [hunk 1]
```diff
@@ -182,7 +182,7 @@ dof : :class:`~openturns.UnsignedInteger`
 
 Returns
 -------
-rSquared : :class:`~openturns.NumericalScalar`
+rSquared : :class:`~openturns.Scalar`
```
Comment (jschueller) [inline python/src/LinearModelAnalysis_doc.i.in]: you can use the python type name here: float

File: python/src/LinearModelStepwiseAlgorithm_doc.i.in [hunk 2]
```diff
@@ -21,7 +21,7 @@ isForward : :class:`~openturns.Bool`
       the boolean value used for the stepwise regression method direction FORWARD and BACKWARD. 
 startIndices : :class:`~openturns.Indices`
      The indices of start model used for the stepwise regression method direction BOTH.
-penalty : :class:`~openturns.NumericalScalar`
+penalty : :class:`~openturns.Scalar`
```
Comment (jschueller) [inline python/src/LinearModelStepwiseAlgorithm_doc.i.in]: here too

File: python/src/LinearModelAnalysis_doc.i.in [hunk 3]
```diff
@@ -192,7 +192,7 @@ rSquared : :class:`~openturns.NumericalScalar`
 
 Returns
 -------
-adjustedRSquared : :class:`~openturns.NumericalScalar`
+adjustedRSquared : :class:`~openturns.Scalar`
```
Comment (jschueller) [inline python/src/LinearModelAnalysis_doc.i.in]: ...

**Comment by sofianehaddad** [comment]
@jschueller done, thanks

**Comment by sofianehaddad** [comment]
@jschueller ok, done

**Comment by regislebrun** [comment]
In the same spirit, there are still some references to the old type names in the source, either as comments or as identifiers:
```
grep -irl "numerical math" lib/src/  | wc -l
```
returns 69,
```
grep -irl "numerical point" lib/src/  | wc -l
```
returns 9 and
```
grep -irl "numerical point" lib/src/  | wc -l
```
returns 26.

**Comment by sofianehaddad** [comment]
@regislebrun true for the OT API, not this module...

**Pull request merged**